docker-compose.dev.yml

```diff
@@ -1,27 +1,82 @@
 services:
+  nginx:
+    image: nginx:latest
+    ports:
+      - 8080:8080
+    configs:
+      - source: nginx_config
+        target: /etc/nginx/nginx.conf
+    networks:
+      - public
+    depends_on:
+      server:
+        condition: service_healthy
+        restart: true
+
   server:
     build:
       context: ./server
       dockerfile: Dockerfile.dev
-    ports:
-      - '5001:5001'
+    # ports:
+    #   - 5001:5001
```

Comment:
we used 5001 because 5000 conflicts with mac os control center

Reply:
we start the whole dev environment by running `docker compose` and there's no need to run a `bun dev` at the same time during development

Reply:
the volume will allow dev docker container to reflect changes
which is also why we don't add healthcheck on dev server itself because when development at some point we can easily make a change that disrupts the server

Reply:
i commented out ports because nginx is exposed and server is accessed through nginx

Reply:
> we start the whole dev environment by running `docker compose` and there's no need to run a `bun dev` at the same time during development

makes sense, but how does it relate to this code

Reply:
> the volume will allow dev docker container to reflect changes which is also why we don't add healthcheck on dev server itself because when development at some point we can easily make a change that disrupts the server

are healthchecks executed everytime we update code? because it wasn't updating for me